lib/gitlab_core.ml

```diff
@@ -1406,6 +1406,18 @@ struct
       API.get ?token ~uri (fun body ->
           return (Gitlab_j.projects_short_of_string body))
 
+    let by_short_ref ?token ~short_ref () =
+      let uri =
+        Uri.of_string
+          (Printf.sprintf "https://gitlab.com/api/v4/projects/%s"
+             (Uri.pct_encode short_ref))
```

Comment:
Could you use the url loaded from the env here https://github.com/tmcgilchrist/ocaml-gitlab/blob/master/lib/gitlab_core.ml#L119? The module is documented here https://github.com/tmcgilchrist/ocaml-gitlab/blob/master/lib/gitlab_s.mli#L1301-L1309

This library should work against the public GitLab.com and any hosted versions of GitLab. Using the env variable for loading the url of the hosted service enables this.

Reply:
Of course, see latest version. I moved the whole `uri` generation into `URI` to a new function `project_by_short_ref`. I was not sure whether I would apply `Uri.pct_encode` in that function or at the call site, because there are no other function in `URI` that takes a path parameter that needs encoding.